File: SampleSolution.Core/Models/ViewModels/Pagination/PagedList.cs
using System.Collections.Generic;

namespace SampleSolution.Core.Models.ViewModels.Pagination
{
    public class PagedList<TEntity>
    {
        public PageMeta PageMeta { get; set; } = new PageMeta();
        public IList<TEntity> Elements { get; set; } = new List<TEntity>();
    }
}

File: SampleSolution.Web/wwwrootsrc/core/scripts/app/models/ViewModels/PagedList.ts
import {PageMeta} from './PageMeta';
import {jsonProperty, Serializable} from 'ts-serializable';

export class PagedList<TEntity> extends Serializable {

    @jsonProperty(PageMeta)
    public pageMeta: PageMeta = new PageMeta();

    @jsonProperty([Object])
    public elements: TEntity[] = [];
}
